handler.go

```diff
@@ -45,22 +44,16 @@ func newErrorHandler(code int, content string) *handler {
 	return h
 }
 
-func (h *handler) Handle(rs []string, w http.ResponseWriter, r *http.Request) {
+func (h *handler) Handle(rs []string, r *http.Request) *response.Response {
 	resp := h.do.Call(
 		h.context(rs, r),
 	)[0].Interface()
 
-	switch resp.(type) {
+	switch v := resp.(type) {
```

Comment:
Nice job, and I think handler does not handle the request anymore, might be a better name would be Dispatcher?

Reply:
How about Invoke?

Reply:
I thinking is still call handler, but use `handler.Invoke` to get the response.

Reply:
I don't think `GetResponder` is a good idea, Get more like just take something all already exists stuff from somewhere, not execute something and get the result.

Reply:
Doesn't that what we did now? Get response.Response this structure then use a detractor wrapping it if detractor is not nil, then use it handle the request.

Reply:
well, `Get` means to me, it means only get something without change anything.
so I don't think Get is appropriate name.